price/priceService.ts

```diff
@@ -1,21 +1,37 @@
-const baseUrl:string = 'https://dydx-testnet.imperator.co/v4';
+import {baseUrl} from "./url.js";
 
-function fetchCandles(pair:string,timeFrame:string):void{
-    fetch(`${baseUrl}/candles/perpetualMarkets/${pair}?resolution=${timeFrame}`)
-    .then(response => {
-        if(!response.ok){
-            throw new Error('Network response was not ok');
-        }
-        else{
-            return response.json();
+type Pair = string;
+type TimeFrame = string;
+
+
+async function fetchData(pair:Pair,timeFrame:TimeFrame): Promise<string> {
+    
+    try {
+        
+        const response = await fetch(`${baseUrl}/candles/perpetualMarkets/${pair}?resolution=${timeFrame}`);
+
+        if (!response.ok) {
+            throw new Error('Network response was not ok or maybe input not valid pair and timeFrame');
         }
-    })
-    .then(data=>{
-        console.log(data);
-    })
-    .catch(error=>{
-        console.error('Error:', error);
-        console.log("Check pair or timeFrame you enter");
-    })
+
+        const data = await response.json();
+
+        return data;
+
+    } catch (error) {
+        console.error('Error fetching data:', error);
```

Comment:
Do not catch errors only for logging them. In other words,
```js
try {
  somethingThatThrows();
} catch (error) {
  console.error(error);
  throw error;
}
```
is a bad practice most of the times, if you only log the error. It's better just not to handle the error at all, and leave it to parent scopes:
```ts
somethingThatThrows(); // not in a try-catch block
```
Error handling should be... error handling. But logging is not handling (most of the times).

Reply:
yes. that's great point 👌 . I remove this part

but I should change function to this ? I think if I don't want to handle error we don't need try too.

async function fetchData(pair:Pair,timeFrame:TimeFrame): Promise<string> {
        
        const response = await fetch(`${baseUrl}/candles/perpetualMarkets/${pair}?resolution=${timeFrame}`);

        if (!response.ok) {
            throw new Error('Network response was not ok or maybe input not valid pair and timeFrame');
        }

        const data = await response.json();

        return data;
}



the past form of function 

async function fetchData(pair:Pair,timeFrame:TimeFrame): Promise<string> {
    
    try {
        
        const response = await fetch(`${baseUrl}/candles/perpetualMarkets/${pair}?resolution=${timeFrame}`);

        if (!response.ok) {
            throw new Error('Network response was not ok or maybe input not valid pair and timeFrame');
        }

        const data = await response.json();

        return data;

    } catch (error) {
        console.error('Error fetching data:', error);
        throw error;
    }

}

Reply:
Yes, you can remove `try/catch` block.